From ad548a350635141c415930c70ce31d8173592a0f Mon Sep 17 00:00:00 2001
From: "Gary D. Gregory" <ggregory@apache.org>
Date: Mon, 23 Jul 2018 15:37:37 +0000
Subject: [PATCH] [VFS-667]
 org.apache.commons.vfs2.provider.res.ResourceFileProvider.findFile(FileObject,
 String, FileSystemOptions) should throw a
 org.apache.commons.vfs2.FileSystemException instead of a NPE when the class
 loader is null.

git-svn-id: https://svn.apache.org/repos/asf/commons/proper/vfs/trunk@1836500 13f79535-47bb-0310-9956-ffa450edef68
---
 .../apache/commons/vfs2/provider/res/ResourceFileProvider.java | 3 +++
 src/changes/changes.xml                                        | 3 +++
 2 files changed, 6 insertions(+)

diff --git a/commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/res/ResourceFileProvider.java b/commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/res/ResourceFileProvider.java
index e77eef2d4a..063c2a4b0b 100644
--- a/commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/res/ResourceFileProvider.java
+++ b/commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/res/ResourceFileProvider.java
@@ -64,6 +64,9 @@ public FileObject findFile(final FileObject baseFile, final String uri, final Fi
         if (cl == null) {
             cl = getClass().getClassLoader();
         }
+        if (cl == null) {
+            throw new FileSystemException("vfs.provider.url/badly-formed-uri.error", uri);
+        }
         final URL url = cl.getResource(resourceName);
 
         if (url == null) {
diff --git a/src/changes/changes.xml b/src/changes/changes.xml
index c1e882eccb..8a9f40fe1b 100644
--- a/src/changes/changes.xml
+++ b/src/changes/changes.xml
@@ -77,6 +77,9 @@ The <action> type attribute can be add,update,fix,remove.
       <action issue="VFS-666" dev="ggregory" type="update">
         Update Apache Commons Collections from 4.1 to 4.2.
       </action>
+      <action issue="VFS-667" dev="ggregory" type="fix">
+        org.apache.commons.vfs2.provider.res.ResourceFileProvider.findFile(FileObject, String, FileSystemOptions) should throw a org.apache.commons.vfs2.FileSystemException instead of a NPE when the class loader is null.
+      </action>
     </release>
     <release version="2.2" date="2017-10-06" description="New features and bug fix release.">
       <action issue="VFS-642" dev="pschumacher" type="update" due-to="ilangoldfeld">
